drivers/meter/ade7754/ade7754.c

```diff
@@ -0,0 +1,1156 @@
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ade7754_adcs_mode(struct ade7754_dev *dev,
+		      enum ade7754_dismod sel)
+{
+	int ret;
+
+	switch (sel) {
+	case ADE7754_POWER_DOWN_MODE:
+		ret = ade7754_update_bits(dev, ADE7754_REG_OPMODE, ADE7754_DISMOD_MSK,
+					  no_os_field_prep(ADE7754_DISMOD_MSK, ADE7754_POWER_DOWN_MODE));
+		break;
+	case ADE7754_OFF_VOLTAGE_ADCS_REDIRECTED_TO_CURRENT:
+		ret = ade7754_update_bits(dev, ADE7754_REG_OPMODE, ADE7754_DISMOD_MSK,
+					  no_os_field_prep(ADE7754_DISMOD_MSK,
+							  ADE7754_OFF_VOLTAGE_ADCS_REDIRECTED_TO_CURRENT));
+		break;
+	case ADE7754_OFF_CURRENT_ADCS_REDIRECTED_TO_VOLTAGE:
+		ret = ade7754_update_bits(dev, ADE7754_REG_OPMODE, ADE7754_DISMOD_MSK,
+					  no_os_field_prep(ADE7754_DISMOD_MSK,
+							  ADE7754_OFF_CURRENT_ADCS_REDIRECTED_TO_VOLTAGE));
+		break;
+	case ADE7754_SWAP_VOLTAGE_CURRENT:
+		ret = ade7754_update_bits(dev, ADE7754_REG_OPMODE, ADE7754_DISMOD_MSK,
+					  no_os_field_prep(ADE7754_DISMOD_MSK, ADE7754_SWAP_VOLTAGE_CURRENT));
+		break;
+	case ADE7754_SLEEP_MODE:
+		ret = ade7754_update_bits(dev, ADE7754_REG_OPMODE, ADE7754_DISMOD_MSK,
+					  no_os_field_prep(ADE7754_DISMOD_MSK, ADE7754_SLEEP_MODE));
+		break;
+	case ADE7754_SWITCH_OFF_VOLTAGE_ADC:
+		ret = ade7754_update_bits(dev, ADE7754_REG_OPMODE, ADE7754_DISMOD_MSK,
+					  no_os_field_prep(ADE7754_DISMOD_MSK, ADE7754_SWITCH_OFF_VOLTAGE_ADC));
+		break;
+	case ADE7754_SWITCH_OFF_CURRENT_ADC:
+		ret = ade7754_update_bits(dev, ADE7754_REG_OPMODE, ADE7754_DISMOD_MSK,
+					  no_os_field_prep(ADE7754_DISMOD_MSK, ADE7754_SWITCH_OFF_CURRENT_ADC));
+		break;
+	default:
+		ret = ade7754_update_bits(dev, ADE7754_REG_OPMODE, ADE7754_DISMOD_MSK,
+					  no_os_field_prep(ADE7754_DISMOD_MSK, ADE7754_NORMAL_OPERATION));
+		break;
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Read energy values
+ * @param dev - The device structure.
+ * @param data - Structure to store energy values
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ade7754_energy_vals(struct ade7754_dev *dev,
+			struct ade7754_energy_values *data)
+{
+	int ret;
+	/* temporary value read from register */
+	uint32_t temp_val;
+
+	if (!data)
+		return -EINVAL;
+
+	/* read accumulated active energy */
+	ret = ade7754_read(dev, ADE7754_REG_AENERGY, &temp_val);
+	if (ret)
+		return ret;
+	data->active_energy_reg_val = (int32_t)temp_val;
+
+	/* read accumulated apparent energy */
+	ret = ade7754_read(dev, ADE7754_REG_VAENERGY, &temp_val);
+	if (ret)
+		return ret;
+	data->apparent_energy_reg_val = (int32_t)temp_val;
+
+	return 0;
+}
+
+/**
+ * @brief Read rms values phase A
+ * @param dev - The device structure.
+ * @param data - Structure to store rms values
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ade7754_rms_vals_phase_a(struct ade7754_dev *dev,
+			     struct ade7754_rms_values *data)
+{
+	int ret;
+	/* temporary value read from register */
+	uint32_t temp_val;
```

Comment:
make sure to apply the uint32_t -> int32_t change that @amiclaus  brought up above to all functions